xwp/wp-core-media-widgets
# Add ability to reset props to defaults when media changes

This branch seeks to fix #106 by adding a new schema attribute `reset_on_media_change` that the callback in `selectMedia()` utilizes to determine if any `model` attributes should be reset back to their default value.

The initial use-case for this is clearing out of the `poster` image when changing videos from a YouTube video, back to a locally hosted video - but there may be other attributes that we determine should be reset back to the default when the media changes.

Here I trigger the logic of media change when the `url` attribute changes, but I almost feel like we should just key off of `attachment_id` instead.... but using `url` handles the case where you are changing between externally hosted videos, and the second one chosen not having a poster image returned in the oembed response  ( edgiest of edge cases there ).

__To Test__
- Add a new video widget
- Set the URL to a YouTube video
- Confirm the poster image is shown in the `renderPreview`
- Click "Change Video"
- Select a video file from the media library
- Note in the `renderPreview` that the old YouTube poster image is not shown.
